### PR TITLE
extension: document extension fields

### PR DESCRIPTION
**Pull request by jrcastro2**
* Add vocabulary mapping for applicability and
  experiments fields
* Updates standard_review:checkdate to a string
* closes https://github.com/CERNDocumentServer/cds-ils/issues/226

### CODE REVIEW

**Review by kprzerwa — COMMENTED** [review]
we wait for the fields numbers to be provided by the library

File: cds_ils/config.py
Comment (jrcastro2) [reply cds_ils/config.py:460]: I didn't check all records but some of them doesn't have a date in the field 925__z. For example, this [record](https://cds.cern.ch/record/2690790/export/xm?ln=en) it's value is `Reviewed in December 2019`

File: cds_ils/vocabularies/data/document_experiments.json
Comment (jrcastro2) [reply cds_ils/vocabularies/data/document_experiments.json:527]: This vocabularies were copied from [here](https://cds.cern.ch/kb/export?kbname=Experiments) but while adding the mapping for the migration I noticed that these values seem duplicated (they are very similar to the previous or equal), that's why I removed them.

File: cds_ils/importer/providers/cds/rules/base.py
Comment (ntarocco) [inline cds_ils/importer/providers/cds/rules/base.py:422]: Is this still valid? If yes, is there an issue for this? Please prefer issues with a tag BLOCKER instead, we will forget all TODOs otherwise

File: cds_ils/importer/providers/cds/rules/values_mapping.py
Comment (ntarocco) [inline cds_ils/importer/providers/cds/rules/values_mapping.py:45]: what is the meaning of these letters? Can we have something more readable? Or are this values coming from the migrated data?

File: cds_ils/importer/providers/cds/rules/base.py
Comment (jrcastro2) [reply cds_ils/importer/providers/cds/rules/base.py:422]: Yes, the librarians are cleaning the values, and when they finish we can add the mapping. From a previous comment from @kprzerwa  I understand that we will wait until the librarians clean the values to merge this PR, so we can add the mapping in this same PR.

File: cds_ils/importer/providers/cds/rules/values_mapping.py
Comment (jrcastro2) [reply cds_ils/importer/providers/cds/rules/values_mapping.py]: Your comment is still valid, maybe @kprzerwa can answer better. I just want to clarify that this is duplicated code that I added wrongly and it's not part of this PR, I already removed it, sorry...

File: cds_ils/config.py
Comment (kprzerwa) [reply cds_ils/config.py:460]: then it should raise an error when we import, because the data is a wrong type. We shouldn't change our new data model because of the inconsistency in the old system

File: cds_ils/importer/providers/cds/rules/base.py
Comment (kprzerwa) [reply cds_ils/importer/providers/cds/rules/base.py:422]: yes, could you please ask them when we could expect the values ?

**Comment by ntarocco** [comment]
Some conflicts, could you please rebase?

File: cds_ils/vocabularies/data/document_insitutions.json
Comment (ntarocco) [inline cds_ils/vocabularies/data/document_insitutions.json:1]: typo in the filename `insitutions`

**Review by ntarocco — COMMENTED** [review]
Did you check if you can edit custom fields in the document form editor?

**Comment by jrcastro2** [comment]
> Did you check if you can edit custom fields in the document form editor?

Yes, we can edit except for the known issue with th `Study` field, it expects a list but it's a text field (this was also happening with `institution` but afet this change now it is a vocabulary field)
image

**Review by ntarocco — APPROVED** [review]

File: cds_ils/importer/providers/cds/rules/base.py
Comment (kprzerwa) [inline cds_ils/importer/providers/cds/rules/base.py:441]: what if we have at some point an institution with two word name ? it will probably not happen for the migration but I wanted to make a point here that it would be safer to use `strip()` since in this way we replace all the spaces

File: tests/migrator/test_books.py
Comment (kprzerwa) [inline tests/migrator/test_books.py:1045]: are there any examples of records from legacy with this date format ?
the tests were written to mirror the real records, f.e. https://cds.cern.ch/record/2690790
I am asking because in this case we have to somehow normalise the data

File: tests/migrator/test_books.py
Comment (kprzerwa) [inline tests/migrator/test_books.py:1114]: shouldn't be only `SPS`?

Comment (jrcastro2) [reply tests/migrator/test_books.py:1114]: I kept it as `CERN SPS` because we got the values from the [accelerators KB](https://cds.cern.ch/kb/export?kbname=Accelerators).

File: tests/migrator/test_books.py
Comment (jrcastro2) [reply tests/migrator/test_books.py:1045]: Your are right, I checked the records and all of them have `Reviewed in XXX`. I will normalise it, thanks!

File: tests/migrator/test_books.py
Comment (kprzerwa) [reply tests/migrator/test_books.py:1114]: yes I see but the goal was to separate the accelerators/experiments from institution, so it is an inconsistency with the rest

Comment (jrcastro2) [reply tests/migrator/test_books.py:1114]: Ok, I will remove `CERN` from the accelerators and leave only `SPS`. Thanks!

**Review by kprzerwa — APPROVED** [review]

**Pull request merged**
